src/main/java/UI/CheckoutPage.java

```diff
@@ -11,84 +11,89 @@
-        form.add(expiration);
-        jtExpiration.setBounds(100, 20, 165, 25);
-        form.add(jtExpiration);
+        usernameInfo = new LabelTextPanel(usernameLabel, jtUsername);
+        nameInfo = new LabelTextPanel(nameLabel, jtName);
+        cardNumberInfo = new LabelTextPanel(cardNumberLabel, jtCardNumber);
+        cvvInfo = new LabelTextPanel(cvvLabel, jtUsername);
+        expirationInfo = new LabelTextPanel(expirationLabel, jtExpiration);
+        addressInfo = new LabelTextPanel(addressLabel, jtAddress);
 
-        JLabel address = new JLabel("Address");
-        address.setBounds(10, 20, 80, 25);
-        form.add(address);
-        jtAddress.setBounds(100, 20, 165, 25);
-        form.add(jtAddress);
+        this.add(titleLabel);
+        this.add(usernameInfo);
+        this.add(nameInfo);
+        this.add(cardNumberInfo);
+        this.add(cvvInfo);
+        this.add(expirationInfo);
+        this.add(addressInfo);
+        this.add(SUBMIT);
+    }
 
-        SUBMIT = new JButton("Submit");
-        SUBMIT.addActionListener(this);
-        SUBMIT.setBounds(10, 80, 80, 25);
-        form.add(SUBMIT);
+    private void setUpLayout() {
+        LAYOUT.putConstraint(SpringLayout.WEST, titleLabel, 500, SpringLayout.WEST, this);
+        LAYOUT.putConstraint(SpringLayout.WEST, usernameInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.WEST, nameInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.WEST, cardNumberInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.WEST, cvvInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.WEST, expirationInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.WEST, addressInfo, 0, SpringLayout.WEST, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.EAST, SUBMIT, 0, SpringLayout.EAST, addressInfo);
 
-        frame.setVisible(true);
+        //Align everything vertically
+        LAYOUT.putConstraint(SpringLayout.NORTH, titleLabel, 150, SpringLayout.NORTH, this);
+        LAYOUT.putConstraint(SpringLayout.NORTH, usernameInfo, 30, SpringLayout.SOUTH, titleLabel);
+        LAYOUT.putConstraint(SpringLayout.NORTH, nameInfo, 30, SpringLayout.SOUTH, usernameInfo);
+        LAYOUT.putConstraint(SpringLayout.NORTH, cardNumberInfo, 30, SpringLayout.SOUTH, nameInfo);
+        LAYOUT.putConstraint(SpringLayout.NORTH, cvvInfo, 30, SpringLayout.SOUTH, cardNumberInfo);
+        LAYOUT.putConstraint(SpringLayout.NORTH, expirationInfo, 30, SpringLayout.SOUTH, cvvInfo);
+        LAYOUT.putConstraint(SpringLayout.NORTH, addressInfo, 30, SpringLayout.SOUTH, expirationInfo);
+        LAYOUT.putConstraint(SpringLayout.NORTH, SUBMIT, 30, SpringLayout.SOUTH, addressInfo);
     }
 
+
     @Override
     public void actionPerformed(ActionEvent e) {
-        if (e.getSource() == this.SUBMIT) {
+        if (e.getSource() == SUBMIT) {
```

Comment:
i see you have validation written in your form (which still assumes that i'm inputting a number btw), but i don't think it's ever called? this might be why i'm getting a runtime error
<img width="987" alt="Screen Shot 2022-11-21 at 10 01 55 PM" src="https://user-images.githubusercontent.com/36282235/203210846-31edbdbc-116b-45b8-9b98-55f02c4737ff.png">